agent/local/state.go

```diff
@@ -1295,7 +1295,10 @@ func (l *State) deleteService(key structs.ServiceID) error {
 		l.logger.Info("Deregistered service", "service", key.ID)
 		return nil
 
-	case acl.IsErrPermissionDenied(err), acl.IsErrNotFound(err):
+	case acl.IsErrNotFound(err):
+		l.services[key].Token = ""
+		fallthrough
+	case acl.IsErrPermissionDenied(err):
```

Comment:
This works!

What I'd suggest actually is not using the service token for the deregistration. That will simplify the logic and behavior slightly.

Like on [line 1273](https://github.com/hashicorp/consul/pull/14436/files#diff-3f2d3e802e18646676be327dbda91428644785290c9d37d5ca37c2124c4170e7R1273), we could do the following (and same for the deleteCheck function)

```diff
- st := l.aclTokenForServiceSync(key, l.tokens.AgentToken)
+ st := l.tokens.AgentToken
```

What do you think abut this? cc @jkirschner-hashicorp @rboyer